### PR TITLE
A couple of small fixes

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/arch/arm/kernel/thread.c [hunk 1]
```diff
@@ -610,8 +610,13 @@ vaddr_t thread_get_saved_thread_sp(void)
 
 bool thread_addr_is_in_stack(vaddr_t va)
 {
-	struct thread_ctx *thr = threads + thread_get_id();
+	struct thread_ctx *thr;
+	int ct = thread_get_id_may_fail();
+
+	if (ct != -1)
```
Comment (jforissier) [inline core/arch/arm/kernel/thread.c]: `if (ct == -1)`

Comment (jenswi-linaro) [inline core/arch/arm/kernel/thread.c]: Thanks

**Comment by jenswi-linaro** [comment]
Update

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

Note about the (false) checkpatch warnings: I've submitted a patch upstream (https://www.spinics.net/lists/kernel/threads.html#2364905), and a workaround in https://github.com/OP-TEE/optee_os/pull/1123.

**Comment by jenswi-linaro** [comment]
Squashed, rebased, tags applied

**Pull request merged**